packages/webchannel-wrapper/gulpfile.js

```diff
@@ -16,203 +16,65 @@
  */
 
 const gulp = require('gulp');
-const rollup = require('rollup');
-const closureCompiler = require('google-closure-compiler').gulp();
-const del = require('del');
-const path = require('path');
-const sourcemaps = require('gulp-sourcemaps');
-const { resolve } = require('path');
-const commonjs = require('@rollup/plugin-commonjs');
-const rollupSourcemaps = require('rollup-plugin-sourcemaps');
-const typescriptPlugin = require('rollup-plugin-typescript2');
-const typescript = require('typescript');
-const pkg = require('./package.json');
+const concat = require('gulp-concat');
 
-// Copied from "../../scripts/build/rollup_emit_module_package_file" which is ESM
-// and would require converting this file to MJS to use
-function emitModulePackageFile() {
```

Comment:
None of the blobs are ES Modules (they are all CJS) - I'm a bit surprised you can remove this.

Isn't the webchannel-wrapper package  part of the public deps tree for Firebase? If you drop this, won't that package only contain CJS inputs and might fail for customers who expect all the packages they depend on to have ES modules?

Reply:
I didn't realize the _es2018.js files were CJS. I think we do need to have ESM exports, and they should be in the "default" fields in package.json, while these cjs files should be in the "require" fields. If ESM versions are not to be provided in closure-net, we'll have to create them here out of the CJS files. Unfortunately we can't just use the previous build process, as it used the closure compiler (which is why it uses gulp instead of just command-line rollup). Maybe the best way to go about this would be, once this PR is modified to use NPM deps instead of submodules, I can make a PR to this branch replacing gulp with rollup and setting the proper package.json paths. This is the only package that still uses gulp (because it was needed for closure compiler) and it would be great to get rid of it and use the same rollup pipeline as all the other packages.

Reply:
> Unfortunately we can't just use the previous build process, as it used the closure compiler (which is why it uses gulp instead of just command-line rollup)

You can do the same thing you do today with rollup, just providing the relevant source file directly instead of generating it using the compiler. In other words, any usage of closureBuild in this build process should be replaced with a direct read of the corresponding file from closure-net. You'd still need the rollup CommonJS plugin.

Reply:
This is a mistake on my part. I'll reach out to @hsubox76 for next steps.

Reply:
Fixed by @hsubox76 via https://github.com/firebase/firebase-js-sdk/pull/8203